### PR TITLE
Secondary HANA stop test

### PR DESCRIPTION
**Pull request by mpagot**
Add test to stop secondary
Cluster OS user controlled by the test
Fix in Trento deployment from git about branches not available in the repo present in the qcow2
Remove more $self in Trento lib
Cypress to support Agent version expectation
More clusters destroy in case of failure

- Verification run: https://openqaworker15.qa.suse.cz/tests/64356#

### CODE REVIEW

File: tests/sles4sap/trento/install_agents.pm [hunk 1]
```diff
@@ -31,7 +31,7 @@ sub run {
         }
     }
 
-    $cmd = $self->install_agent($wd, '/root/test', $agent_api_key, '10.0.0.4');
+    $cmd = trento::install_agent($wd, '/root/test', $agent_api_key, '10.0.0.4');
```
Comment (jankohoutek) [inline tests/sles4sap/trento/install_agents.pm:34]: IP is hard coded for which reason?

**Review by jankohoutek — COMMENTED** [review]
See inline comments....

File: tests/sles4sap/trento/test_trento_deploy.pm [hunk 2]
```diff
@@ -10,7 +10,7 @@ use Mojo::Base 'publiccloud::basetest';
 use base 'consoletest';
 use testapi;
 use serial_terminal 'select_serial_terminal';
-use utils qw(script_retry);
+use utils 'script_retry';
```
Comment (alvarocarvajald) [inline tests/sles4sap/trento/test_trento_deploy.pm:13]: This is not necessary: https://github.com/os-autoinst/os-autoinst/pull/1375

But you can leave it if you want.

File: tests/sles4sap/trento/test_trento_deploy.pm [hunk 3]
```diff
@@ -21,7 +21,7 @@ sub run {
 
     my $machine_ip = $self->get_trento_ip;
     if (!get_var('TRENTO_EXT_DEPLOY_IP')) {
-        my $resource_group = $self->get_resource_group;
+        my $resource_group = trento::get_resource_group;
```
Comment (alvarocarvajald) [inline tests/sles4sap/trento/test_trento_deploy.pm:24]: Same question regarding `use base 'trento';` and `trento::`.

File: tests/sles4sap/trento/test_trento_deploy.pm [hunk 4]
```diff
@@ -54,7 +54,7 @@ sub post_fail_hook {
     my ($self) = @_;
     if (!get_var('TRENTO_EXT_DEPLOY_IP')) {
         trento::k8s_logs(qw(web runner));
-        $self->az_delete_group;
+        trento::az_delete_group;
```
Comment (alvarocarvajald) [inline tests/sles4sap/trento/test_trento_deploy.pm:57]: Same question regarding `use base 'trento';` and `trento::`.

File: tests/sles4sap/trento/test_trento_web.pm [hunk 5]
```diff
@@ -38,14 +38,15 @@ sub run {
 sub post_fail_hook {
     my ($self) = @_;
     if (!get_var('TRENTO_EXT_DEPLOY_IP')) {
-        $self->az_delete_group;
+        trento::k8s_logs(qw(web runner));
```
Comment (alvarocarvajald) [inline tests/sles4sap/trento/test_trento_web.pm:41]: Same question regarding `use base 'trento';` and `trento::`.

File: tests/sles4sap/trento/test_trento_web.pm [hunk 6]
```diff
@@ -38,14 +38,15 @@ sub run {
...
+        trento::az_delete_group;
```
Comment (alvarocarvajald) [inline tests/sles4sap/trento/test_trento_web.pm:42]: Same question regarding `use base 'trento';` and `trento::`.

File: tests/sles4sap/trento/test_trento_web.pm [hunk 7]
```diff
@@ -38,14 +38,15 @@ sub run {
...
     }
 
     $self->cypress_log_upload(('.txt', '.mp4'));
     parse_extra_log("XUnit", $_) for split(/\n/, script_output('find ' . $self->CYPRESS_LOG_DIR . ' -type f -iname "*.xml"'));
 
-    $self->destroy_qesap();
-
+    qesap_upload_logs();
+    trento::destroy_qesap();
```
Comment (alvarocarvajald) [inline tests/sles4sap/trento/test_trento_web.pm:49]: Same question regarding `use base 'trento';` and `trento::`.

**Comment by lpalovsky** [comment]
In the VR I see that the stop Hana test failed.
Did the HANA stop command pass?  I cannot find the command in serial output. 

https://openqaworker15.qa.suse.cz/tests/64356/#step/test_hana_stop/16
This one is strange as well.

**Comment by mpagot** [comment]
> In the VR I see that the stop Hana test failed. Did the HANA stop command pass? I cannot find the command in serial output.
> 
> https://openqaworker15.qa.suse.cz/tests/64356/#step/test_hana_stop/16 This one is strange as well.

no does not pass, as `ansible` is not found... very strange. But take care that the cluster is not there at all anymore as the **test_trento_web** test failed before and the clean up has been triggered https://openqaworker15.qa.suse.cz/tests/64356/#step/test_trento_web/352


The Trento fails as the HANA database is not reported as running, that is what also previously confirmed (but not checked) by the test code here https://openqaworker15.qa.suse.cz/tests/64356/#step/cluster_deploy/54

It is a new issue https://jira.suse.com/browse/TEAM-6936

It is a show stopper for the VR of this PR

File: tests/sles4sap/trento/cluster_deploy.pm [hunk 8]
```diff
@@ -28,6 +28,11 @@ sub run {
         '-a', trento::get_vnet($cluster_rg));
     record_info('NET PEERING');
     assert_script_run($cmd, 360);
+
+    my $inventory = qesap_get_inventory(get_required_var('PUBLIC_CLOUD_PROVIDER'));
+    my $prov = get_required_var('PUBLIC_CLOUD_PROVIDER');
```
[the remark this answers is not in the document]
Comment (alvarocarvajald) [reply tests/sles4sap/trento/cluster_deploy.pm:33]: I like this.

File: tests/sles4sap/trento/install_agents.pm (hunk 1 above)
Comment (mpagot) [reply tests/sles4sap/trento/install_agents.pm]: I do not have a way to dynamically calculate it yet, TODO

File: tests/sles4sap/trento/cluster_configure.pm [hunk 9]
```diff
@@ -20,7 +20,7 @@ sub run {
     my $provider = $self->provider_factory();
 
     # Setup and configure the qe-sap-deployment
-    $self->config_cluster($provider->provider_client->region);
+    trento::config_cluster($provider->provider_client->region);
```
[the remark this answers is not in the document]
Comment (mpagot) [reply tests/sles4sap/trento/cluster_configure.pm]: done as part of https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/15864

File: tests/sles4sap/trento/test_hana_stop.pm [hunk 10]
```diff
@@ -0,0 +1,47 @@
+# Copyright SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+# Summary: Trento stop secondary HANA test
```
[the remark this answers is not in the document]
Comment (mpagot) [reply tests/sles4sap/trento/test_hana_stop.pm:4]: I think so, yes, with interleaved CY test and communicating with the hosts using Ansible

**Comment by mpagot** [comment]
Move to https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/15881

**Pull request closed without merging**